src/WebApiToTypeScript/Enums/EnumsService.cs

```diff
@@ -61,5 +67,41 @@ private void CreateEnumForType(TypeScriptBlock enumsBlock, TypeDefinition typeDe
             foreach (var field in fields)
                 enumBlock.AddStatement($"{field.Name} = {field.Constant},");
         }
+
+        private void CreateGetDescriptionExtensionForType(TypeScriptBlock enumsBlock, TypeDefinition typeDefinition)
+        {
+            var fields = typeDefinition.Fields
+                .Where(f => f.HasConstant && !f.IsSpecialName);
+
+            var switchBlock = enumsBlock
+                .AddAndUseBlock($"export namespace {typeDefinition.Name}")
+                .AddAndUseBlock($"export function getDescription(enumValue: {typeDefinition.Name})")
+                .AddAndUseBlock("switch (enumValue)");
+
+            foreach (var field in fields)
+            {
+                var fieldDescription = GetFieldDescription(field);
+                switchBlock
+                    .AddStatement($"case {typeDefinition.Name}.{field.Name}: return \"{fieldDescription}\";");
+            }
+        }
+
+        private string GetFieldDescription(FieldDefinition field)
+        {
+            var regexToFindUppercases = new Regex(@"
+                (?<=[A-Z])(?=[A-Z][a-z]) |
+                 (?<=[^A-Z])(?=[A-Z]) |
+                 (?<=[A-Za-z])(?=[^A-Za-z])", RegexOptions.IgnorePatternWhitespace);
+
+            var descriptionAttributeName = typeof(DescriptionAttribute).FullName;
+
+            var fieldDescription =
+                field.CustomAttributes.Any(attr => attr.AttributeType.FullName == descriptionAttributeName)
+                    ? field.CustomAttributes.Single(attr => attr.AttributeType.FullName == descriptionAttributeName)
```

Comment:
I think if I wrote this bit, I'd have written .Single .Construc .Value .ToString() all on separate lines ;)